src/Orleans/Providers/IOrleansProvider.cs

```diff
@@ -140,5 +140,25 @@ public static Type GetTypeProperty(this IProviderConfiguration config, string ke
             string s;
             return config.Properties.TryGetValue(key, out s) ? Type.GetType(s) : settingDefault;
         }
+
+        public static bool GetBoolProperty(this IProviderConfiguration config, string key, bool settingDefault)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+            string s;
+            return config.Properties.TryGetValue(key, out s) ? bool.Parse(s) : settingDefault;
+        }
+
+        public static TimeSpan GetTimeSpanProperty(this IProviderConfiguration config, string key, TimeSpan settingDefault)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+            string s;
+            return config.Properties.TryGetValue(key, out s) ? TimeSpan.Parse(s) : settingDefault;
```

Comment:
Behavior needs to be symmetric.  We need to both read and write timespan.  ConfigUtilities only deals with reads.